flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/services.dart';
+import 'package:flutter/material.dart';
+import 'package:sentry/sentry.dart';
+
+typedef _Parser<T> = Future<T> Function(String value);
+
+/// An [AssetBundle] which creates automatic performance traces for loading
+/// assets.
+///
+/// You can wrap other [AssetBundle]s in it:
+/// ```dart
+/// SentryAssetBundle(bundle: someOtherAssetBundle)
+/// ```
+/// If you're not providing any [AssetBundle], it falls back to the [rootBundle].
+///
+/// If you want to use the [SentryAssetBundle] by default you can achieve this
+/// with the following code:
+/// ```dart
+/// DefaultAssetBundle(
+///   bundle: SentryAssetBundle(),
+///   child: MaterialApp(
+///     home: MyScaffold(),
+///   ),
+/// );
+/// ```
+/// [Image.asset], for example, will then use [SentryAssetBundle].
+class SentryAssetBundle extends AssetBundle {
+  SentryAssetBundle({Hub? hub, AssetBundle? bundle})
+      : _hub = hub ?? HubAdapter(),
+        _bundle = bundle ?? rootBundle;
+
+  final Hub _hub;
+  final AssetBundle _bundle;
+
+  @override
+  Future<ByteData> load(String key) async {
+    final span =
+        _hub.getSpan()?.startChild('SentryAssetBundle.load', description: key);
+
+    try {
+      final data = await _bundle.load(key);
+      await span?.finish();
+      return data;
+    } catch (e) {
+      await span?.finish(status: SpanStatus.internalError());
+      rethrow;
+    }
+  }
+
+  @override
+  Future<T> loadStructuredData<T>(
+    String key,
+    _Parser<T> parser,
+  ) async {
+    final span = _hub.getSpan()?.startChild(
+          'SentryAssetBundle.loadStructuredData',
```

Comment:
I'm not sure what to use for name and operation

Reply:
According to https://develop.sentry.dev/sdk/performance/span-operations/ it should probably be `file.read` as a name and `AssetBundle.loadStructuredData($filename)` as description. Or should the filename be added as a tag?

Reply:
@romtsn

Reply:
@ueman descriptions are still not following @romtsn comment from the other thread: https://github.com/getsentry/sentry-dart/pull/685#discussion_r786022279

See here: https://github.com/getsentry/sentry-java/blob/4c1cf85df7af23b88495120bd9a089e601593eef/sentry/src/main/java/io/sentry/instrumentation/file/FileIOSpanManager.java#L86-L94
It's supposed to be something like, e.g. `file.txt (1 MB)`

Reply:
We don't have any code to pretty print the file size in the SDK. Any suggestions for that?

Reply:
@ueman based on https://github.com/getsentry/develop/issues/463#issuecomment-990042948 just the last path segment in the description and the while path in the `file.path` data field (already correct).
For the file size, can we do something like https://gist.github.com/zzpmaster/ec51afdbbfa5b2bf6ced13374ff891d9#file-formatbytes-dart ?

Reply:
I've just noticed that we can't easily modify the spans description after it was started.
So I can add the file name, but I can't add the file size.

Reply:
That sounds like a good start, we can iterate that on a different PR.